packages/core/src/fhir-to-cda/cda-generators.ts

```diff
@@ -0,0 +1,55 @@
+import { Bundle } from "@medplum/fhirtypes";
+import { findOrganizationResource, findPatientResource } from "../external/fhir/shared";
+import BadRequestError from "../util/error/bad-request";
+import { buildAuthor } from "./cda-templates/clinical-document/author";
+import { buildClinicalDocumentXML } from "./cda-templates/clinical-document/clinical-document";
+import { buildCustodian } from "./cda-templates/clinical-document/custodian";
+import { buildRecordTargetFromFhirPatient } from "./cda-templates/clinical-document/record-target";
+import { buildStructuredBody } from "./cda-templates/clinical-document/structured-body";
+
+export function generateCdaFromFhirBundle(fhirBundle: Bundle): string {
+  const patientResource = findPatientResource(fhirBundle);
+  const organizationResources = findOrganizationResource(fhirBundle);
+
+  if (!patientResource || !organizationResources) {
+    const missing = [];
+    if (!patientResource) {
+      missing.push("Patient");
+    }
+    if (!organizationResources) {
+      missing.push("Organization");
+    }
+    throw new BadRequestError(`${missing.join(", ")} resource(s) not found`);
+  }
+
+  const recordTarget = buildRecordTargetFromFhirPatient(patientResource);
+  const author = buildAuthor(organizationResources);
+  const custodian = buildCustodian();
+  const structuredBody = buildStructuredBody(fhirBundle);
+
+  if (!recordTarget || !author || !custodian || !structuredBody) {
+    const missing = [];
+    if (!recordTarget) {
+      missing.push("recordTarget");
+    }
+    if (!author) {
+      missing.push("author");
+    }
+    if (!custodian) {
+      missing.push("custodian");
+    }
+    if (!structuredBody) {
+      missing.push("structuredBody");
+    }
+
+    throw new BadRequestError(`${missing.join(", ")} resource(s) not found`);
```

Comment:
we are saying "resources" not found here, but the things we are putting into missing are CDA components and not FHIR resources. Could just change the wording to "Components"? What is the case where patient + organization but dont get recordTarget, author, custodian, etc.

Reply:
Good point. I'll adjust this part of the code. recordTarget, author and custodian will never be undefined here